### PR TITLE
feat(Datatable): add selection features

### PR DESCRIPTION
**Pull request by ajkl2533**
Closes FEP-1563, FEP-1565

### CODE REVIEW

File: src/components/forms/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -1,36 +1,50 @@
...
 import { FlexContainer } from '../../FlexContainer';
 import { Icon } from '../../Icon';
-import { IconTypes, SSCIconNames } from '../../../theme/icons/icons.enums';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
 import { Label } from '../Label';
 import { TogglingInputProps } from '../types/forms.types';
 import { CheckboxProps } from './Checkbox.types';
 
 const CheckboxWrapper = styled(FlexContainer)`
+  position: relative;
```
Comment (PrincAm) [inline src/components/forms/Checkbox/Checkbox.tsx:22]: Pls, why relative?

Comment (ajkl2533) [reply src/components/forms/Checkbox/Checkbox.tsx:22]: forgot to remove it :D

File: src/components/Datatable/Table/SelectionCheckbox/SelectionCheckbox.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { forwardRef, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+import { Checkbox } from '../../../forms';
+import { SelectionCheckboxProps } from './SelectionCheckbox.types';
+
+// Explanation: https://itnext.io/reusing-the-ref-from-forwardref-with-react-hooks-4ce9df693dd
```
Comment (PrincAm) [inline src/components/Datatable/Table/SelectionCheckbox/SelectionCheckbox.tsx:7]: 👍

**Review by PrincAm — APPROVED** [review]
LGTM

**Review by youssefssc — APPROVED** [review]
LGTM

**Pull request merged**